backend/app/controllers/spree/admin/cancellations_controller.rb

```diff
@@ -36,6 +36,8 @@ def created_by
       def load_order
         @order = Spree::Order.find_by!(number: params[:order_id])
         authorize! action, @order
+      rescue ActiveRecord::RecordNotFound => e
+        resource_not_found(flash_class: e.model.constantize, redirect_url: admin_orders_path)
```

Comment:
Why are you not passing the error to `resource_not_found` here? I saw the commit message but it's still not entirely clear.

Reply:
If I understand your question correctly, you asking why I did not pass `e` into the `_error` of the resource_not_found of the base_controller?
I thought it is not necessary as I am not using the `_error` in the base_controller#resource_not_found. The `_error` is needed because some controllers use the `rescue_from` method which passes the exception object to the handler if it accepts an argument. This results in an `ArgumentError ... wrong number of arguments (given 1, expected 0)` if `_error` is not there.

On that note, I overlooked the fact that I do not even need the `e` at all in this cancellations_controller rescue. I should just pass `Spree::Orders` instead of unnecessarily passing `e.model.contantize` for the flash_class, since `e.model.constantise` will always return `Spree::Order`. Oddly enough, I seem to have appropriately done this in the other controller rescue methods but overlooked this one. I will make the change.

Reply:
@ikraamg how about this, then?
```ruby
rescue_from ActiveRecord::RecordNotFound do
  resource_not_found
end
```
It looks weird to have a method parameter that is not used at all in the method.

Reply:
To keep the format similar to how the rescue is done in the other controllers, I thought this would work for line 39 and 40:

```ruby
rescue ActiveRecord::RecordNotFound
  resource_not_found(flash_class: Spree::Order, redirect_url: admin_orders_path)
```